src/features/pilot_management/store/index.ts

```diff
@@ -165,6 +184,7 @@ export class PilotManagementStore extends VuexModule {
 
   @Action
   public addPilot(payload: { pilot: Pilot; update: boolean }): void {
+    this.context.commit(ADD_GROUP, payload.pilot.Group)
```

Comment:
This is the major piece that will fix #1650: Simply making sure the group is appended to PilotGroups prior to adding the Pilot themselves (and subsequently sorting/reindexing as appropriate).